### PR TITLE
Implements Pollen Puff

### PR DESCRIPTION
**Pull request by lucfd**
Made new attr `ConditionalHealAttr` that gives a move the following properties when a provided callback function returns `true`:
- power is set to zero
- target heals for 50% of their health

### CODE REVIEW

File: src/data/move.ts [hunk 1]
```diff
@@ -1947,6 +1947,29 @@ export class PresentPowerAttr extends VariablePowerAttr {
...
+    this.callback = callback;
+    this.healRatio = healRatio;
+  }
+ 
+  apply(user: Pokemon, target: Pokemon, move: Move, args: any[]): boolean {
+
+    if (this.callback(user, target, move)) {
+      (args[0] as Utils.NumberHolder).value = 0;
+      target.scene.unshiftPhase(new PokemonHealPhase(target.scene, target.getBattlerIndex(),
+      Math.max(Math.floor(target.getMaxHp() * this.healRatio), 1), getPokemonMessage(target, ' regained\nhealth!'), true));
```
Comment (TempsRay) [inline src/data/move.ts:1965]: nit: tab line 1965 so it's clear it's still part of the previous line

File: src/data/move.ts [hunk 2]
```diff
@@ -5437,8 +5460,8 @@ export function initMoves() {
     new AttackMove(Moves.THROAT_CHOP, Type.DARK, MoveCategory.PHYSICAL, 80, 100, 15, 100, 0, 7)
       .partial(),
     new AttackMove(Moves.POLLEN_PUFF, Type.BUG, MoveCategory.SPECIAL, 90, 100, 15, -1, 0, 7)
-      .ballBombMove()
-      .partial(),
+      .attr(ConditionalHealAttr, (user, target, move) => (!user.isPlayer() && !target.isPlayer()) || (user.isPlayer() && target.isPlayer()) ? true : false, 0.5)
```
Comment (TempsRay) [inline src/data/move.ts:5463]: nit: wrapping the whole right side of the => in parentheses to make it clearer
`.attr(ConditionalHealAttr, (user, target, move) => ((!user.isPlayer() && !target.isPlayer()) || (user.isPlayer() && target.isPlayer())) ? true : false, 0.5)`

**Review by TempsRay — COMMENTED** [review]
LGTM

**Review by bennybroseph — CHANGES_REQUESTED** [review]
How does this interact with Telepathy and Heal Block?

**Comment by bennybroseph** [comment]
Closing this due to inactivity. Feel free to make a new request if interest is reignited

**Pull request closed without merging**